### ISSUE TITLE
Authbackend creation fails with path already in use at oidc

### ISSUE THREAD

**Issue by mustafaStakater** [upbound/provider-vault#25]
### What happened? 

Creating AuthBackend fails initially with the `both 'oidc_client_id' and 'oidc_client_secret' must be set for OIDC` and then error changes to `* path is already in use at oidc/` . In vault, `oidc` path is present with empty configuration.

### How can we reproduce it?

- Get a openshift cluster and Deploy vault and crossplane on openshift.
- Access pods cli and Initialize vault using `vault operator init`. Save the unseal keys and root token. 
- Login to vault with root token using `vault login` and create a token for crossplane.
- Create a crossplane provider for vault. 
  ```
  apiVersion: pkg.crossplane.io/v1
  kind: Provider
  metadata:
    name: provider-vault
  spec:
    package: 'xpkg.upbound.io/upbound/provider-vault:v0.1.0'
    controllerConfigRef:
      name: vault-controller
  ---
  apiVersion: pkg.crossplane.io/v1alpha1
  kind: ControllerConfig
  metadata:
    name: vault-controller
  spec:
    securityContext: {}
    podSecurityContext: {}
    args:
      - "--leader-election"
    replicas: 1
  ```
- Create a provider config and secret with token to vault
  ```
  apiVersion: vault.upbound.io/v1beta1
  kind: ProviderConfig
  metadata:
    name: provider-vault
  spec:
    address: 'http://vault.apps-crc.testing/'
    credentials:
      secretRef:
        key: credentials
        name: vault-creds
        namespace: crossplane
      source: Secret
  ---
  kind: Secret
  apiVersion: v1
  metadata:
    name: vault-creds
    namespace: crossplane
  data:
    credentials: "hvs.alshdabuoac"
  type: Opaque
  
  ```
- Create authbackend resource
  ```
  apiVersion: jwt.vault.upbound.io/v1alpha1
  kind: AuthBackend
  metadata:
    name: vault-config-fb77r
  spec:
    deletionPolicy: Delete
    forProvider:
      defaultRole: defaultrole
      oidcClientId: vault
      oidcClientSecretSecretRef:
        key: secret
        name: vault-sso-client-secret
        namespace: vault
      oidcDiscoveryUrl: >-
        https://sso-/auth/realms/vault-realm
      path: oidc
      type: oidc
    managementPolicy: FullControl
    providerConfigRef:
      name: provider-vault
  ---
  kind: Secret
  apiVersion: v1
  metadata:
    name: vault-sso-client-secret
    namespace: vault
  data:
    secret: ckFuRG9NMTIz
  type: Opaque
  
  ```
### What environment did it happen in?
- Crossplane and Vault running in Openshift
- Vault Version : 1.12.1
- Crossplane Version : 1.14.5

**Comment by mustafaStakater** [comment]
The issue was a wrong secret name being used, but provider shouldnt partially create an empty oidc configuration

**Comment by uwefreidank** [comment]
Also happen with
- Vault Version 1.15.7
- Crossplane Version: 1.15.2
- Provider vault: 0.0.4

Even worse, if i apply several AuthBackends in parallel (in my case, 55), the 1st reconciled AuthBackend can successfully be created in Vault, but all other AuthBackends are failing with `path is already in use at oidc`.
Work around: applying the AuthBackends CRs one by one and waiting in between.

**Comment by jonasz-lasut** [comment]
Hi @mustafaStakater thank you for opening the issue!

Closing this as a duplicate of #83, which tracks the same root cause with full provider and Vault logs.

**What happens.** `AuthBackend.jwt` enables the auth mount first and writes `auth/<path>/config` second. In your first report the config write failed because the client secret reference pointed at a Secret that did not exist, so the mount stayed behind with an empty configuration. On the provider version you used, the resource did not record the mount as its own after that failure, so the next reconcile tried to create it again and Vault answered `path is already in use at oidc/`. Current releases (v3.0.0 and later) record the mount before writing the configuration, so this particular case now recovers on its own: the next reconcile finds the empty mount, treats the missing configuration as drift and writes it once the referenced Secret exists.

The parallel case @uwefreidank describes is different and still open. When many backends are created at once and the mount already exists, the create fails before the resource has recorded anything, so every later reconcile tries to create again and fails the same way. Every resource in this provider learns its identity only from a successful create, which is exactly the problem #83 tracks for mounts, audit devices, auth backends and policies.

**Workaround today.** Tell the provider which mount it owns by setting the external name to the path:

```
kubectl annotate authbackend.jwt.vault.upbound.io <name> crossplane.io/external-name=oidc
```

The next reconcile reads the existing mount, writes any missing configuration and marks the resource Ready. Applying the backends one at a time avoids the parallel case but is unnecessary once the annotation is set.

**The fix.** Path-keyed resources should derive their external name from `path` so that Observe finds an existing mount before Create runs and adopts it instead of failing. That change is tracked in #83.
